drivers/npmx/shell/charger.c

```diff
@@ -75,9 +93,36 @@ static int cmd_charger_charging_current_get(const struct shell *shell, size_t ar
 	return 0;
 }
 
+static bool charger_die_temp_set_helper(const struct shell *shell, npmx_charger_t *charger_instance,
+					int16_t temperature, int16_t *p_temperature,
+					npmx_error_t (*func_set)(npmx_charger_t const *p_instance,
+								 int16_t temperature),
+					npmx_error_t (*func_get)(npmx_charger_t const *p_instance,
+								 int16_t *p_temperature))
+{
+	npmx_error_t err_code = func_set(charger_instance, temperature);
+	if (!check_error_code(shell, err_code)) {
+		print_set_error(shell, "die temperature threshold");
+		return false;
+	}
+
+	err_code = func_get(charger_instance, p_temperature);
+	if (!check_error_code(shell, err_code)) {
+		print_get_error(shell, "die temperature threshold");
+		return false;
+	}
+
+	if (temperature != *p_temperature) {
+		shell_info(shell, "Info: Set value may be different than requested.");
```

Comment:
I made calculations and tests - for all valid values of die_temp (from -416 *C to +395 *C), get returns exactly the same value which was set.